src/middleware/fileupload.js

```diff
@@ -1,56 +1,60 @@
-                fs.rename(oldpath, newpath, function(err) {
-                    if (err) throw err;
-                    else {
-                        var lineReader = require('readline').createInterface({
-                            input: fs.createReadStream(newpath)
-                        });
-                        var count = 0;
-                        var line_array = [];
-                        var filtered_data = [];
+    fileUpload: function(req, res) {
+        return new Promise((resolve, reject) => {
+            var oldpath = config.path;
+            var newpath = path.join(__dirname, './uploads/AGL_001.TXT');
+            fs.rename(oldpath, newpath, function(err) {
+                if (err) throw err;
+                else {
+                    var lineReader = require('readline').createInterface({
+                        input: fs.createReadStream(newpath) //stores file to ne path in upload
+                    });
+                    var count = 0;
+                    var line_array = [];
+                    var filtered_data = [];
+
+                    function readFile(callback) {
+                        lineReader.on('line', function(line) {
+                            line_array.push(line.split("\t"))
+                            if (line_array[count][0] && count >= 1) {
+                                console.log(line_array[count])
+                                var fetched_data = {
+                                    id: line_array[count][0],
+                                    user_id: line_array[count][2],
+                                    timing: line_array[count][6]
+                                }
+                                var flag = 0;
+                                _.forEach(filtered_data, function(value) {
 
-                        function readFile(callback) {
-                            lineReader.on('line', function(line) {
-                                line_array.push(line.split("\t"))
-                                if (line_array[count][0] && count >= 1) {
-                                    var fetched_data = {
-                                        No: line_array[count][0],
-                                        TMNo: line_array[count][1],
-                                        EnNo: line_array[count][2],
-                                        Name: line_array[count][3],
-                                        GMNo: line_array[count][4],
-                                        Mode: line_array[count][5],
-                                        "IN/OUT": line_array[count][6],
-                                        Antipass: line_array[count][7],
-                                        DateTime: line_array[count][8]
+                                    if (value.id == fetched_data.id || (fetched_data.user_id).toString().trim() == "0") {
+                                        flag = 1;
```

Comment:
can u explain the logic used here in comments in proper detail. 
so that i or any other person can understand it. 
why are we removing duplicates

Reply:
it checks if id is duplicate ,if yes it will not be stored as id is primary key and will cause validation err at insert query and also check if user id is valid or not